src/server/task/server/api_server.go

```diff
@@ -0,0 +1,69 @@
+package server
+
+import (
+	"path"
+	"strings"
+
+	col "github.com/pachyderm/pachyderm/v2/src/internal/collection"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/taskapi"
+
+	etcd "go.etcd.io/etcd/client/v3"
+)
+
+type apiServer struct {
+	ppsPrefix, pfsPrefix string
+	etcdClient           *etcd.Client
+}
+
+func NewAPIServer(base, pps, pfs string, etcdClient *etcd.Client) taskapi.APIServer {
```

Comment:
I think we should just take an env here and access the prefix fields when we need them rather than storing them in the api server. Also, etcd can be accessed from the env as well.